crates/chain/src/spk_client.rs

```diff
@@ -193,6 +193,12 @@ impl<I> SyncRequestBuilder<I> {
         self
     }
 
+    /// Set the `time_of_sync` for unconfirmed transactions.
+    pub fn time_of_sync(mut self, time_of_sync: u64) -> Self {
```

Comment:
Is this value required ? If so should it be a parameter of `SyncRequest::builder()`?  Otherwise if someone forgets to set it it will always be 0. (And same for FullScanRequest::builder())

Reply:
At first I thought to make it user-configurable, but I think you're right, it makes significantly more sense to just have the `time_of_sync` set by default.

Reply:
Now that you have internal `time_of_sync` field as an `Option<u64>` it makes sense to leave this function as-is.